compiler/rustc_codegen_ssa/src/sir.rs

```diff
@@ -215,6 +227,20 @@ impl SirFuncCx<'tcx> {
         this
     }
 
+    /// Compute layout and offsets required for blackholing.
+    pub fn compute_layout_and_offsets(&mut self) {
+        let mut layout = Layout::from_size_align(0, 1).unwrap();
+        for ld in &self.func.local_decls {
+            let (size, align) = self.layout_map.get(&ld.ty).unwrap();
+            let l = Layout::from_size_align(*size, *align).unwrap();
+            let (nl, off) = layout.extend(l).unwrap();
```

Comment:
couldn't you assign directly to `layout` in the first element of the return tuple?

Reply:
I assume because of the `let` it would create a new variable instead of overwriting the existing one.

Reply:
Ah yes. My bad.